odoo/models.py

```diff
@@ -2986,6 +2986,7 @@ def _field_to_sql(self, alias: str, fname: str, query: (Query | None) = None, fl
             sql_field = SQL('(%s->>0)::%s', sql_field, SQL(field._column_type[1]))
 
             if field.type == 'many2one':
+                assert query
```

Comment:
bool is overwritten, you probably don't want to execute it
```suggestion
                assert query is not None
```